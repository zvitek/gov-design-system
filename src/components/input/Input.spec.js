import { shallowMount, mount } from '@vue/test-utils'
import GovInput from '@components/input/Input'

let wrapper

describe('GovInput', () => {
    beforeEach(() => {
        wrapper = shallowMount(GovInput)
    })

    it('render correctly', () => {
        const wrapper = shallowMount(GovInput, {
            propsData: {
                customUniqueId: 'input_abc'
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('is vue instance', () => {
        expect(wrapper.name()).toBe('GovInput')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('renders input element by default', () => {
        expect(wrapper.contains('input')).toBeTruthy()
        expect(wrapper.classes()).toContain('gov-form-control__input')
    })

    it('render textarea element when type is textarea', () => {
        wrapper.setProps({ textarea: true })
        const target = wrapper.find('textarea')

        expect(target.exists()).toBeTruthy()
    })

    it('render field password when the type property is password', () => {
        const wrapper = shallowMount(GovInput, {
            propsData: {
                type: 'password'
            }
        })

        const target = wrapper.find('input')
        expect(target.exists()).toBeTruthy()
        expect(target.attributes().type).toBe('password')
    })

    it('render the placeholder and readonly attribute when passed', () => {
        const wrapper = shallowMount(GovInput, {
            attrs: { placeholder: 'Awesome!', readonly: true }
        })
        const target = wrapper.find('input')

        expect(target.element.getAttribute('placeholder')).toBe('Awesome!')
        expect(target.element.getAttribute('readonly')).toBe('readonly')
    })

    it('keep its value on blur', async () => {
        const wrapper = mount(GovInput, {
            propsData: {
                value: 'foo'
            }
        })

        const input = wrapper.find('input')

        input.element.value = 'bar'
        input.trigger('input')
        input.trigger('blur')

        expect(input.element.value).toBe('bar')
    })
})
