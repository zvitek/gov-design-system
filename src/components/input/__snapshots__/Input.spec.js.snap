// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovInput render correctly 1`] = `<input type="text" autocomplete="on" class="gov-form-control__input" id="input_abc">`;
